src/checkpoint.rs

```diff
@@ -26,60 +26,31 @@ pub struct CheckpointCreateResult {
 }
 
 impl Db {
-    /// Creates a checkpoint of the db stored in the object store at the specified path using the
-    /// provided options. Note that the scope option does not impact the behaviour of this method.
-    /// The checkpoint will reference the current active manifest of the db.
+    /// Creates a checkpoint of an opened db using the provided options. Returns the ID of the created
+    /// checkpoint and the id of the referenced manifest.
     pub async fn create_checkpoint(
-        path: &Path,
-        object_store: Arc<dyn ObjectStore>,
+        &self,
```

Comment:
Nice, this is a great improvement! I was wondering how you wanted to implement force flush without having write access to the database. This also improves usability quite a bit, I did some prototyping during the weekend and the old API felt bit clumsy (i needed to do additional bookkeeping of object_store).